### DISCUSSION TITLE
Sugestão

### DISCUSSION THREAD

**Discussion by mborgues** [HashLoad/horse#196] [Ideas]
Olá, Gostaria muito de parabenizar os responsáveis pelo maravilhoso projeto e torço muito pela continuidade e evolução framework.
Iniciei a algumas semanas o estudo do framework e os middlewares, mas um em especial me chamou muito a atenção para algo que seria muito interessante ser nativo no Horse, o GBSwagger, ele possui implementações na forma de registro de rotas, controllers e data annotations que se disponibilizados no framework traria mais versatilidade, organização e robustez a ele, assim como mais recursos disponibilizados aos middlewares, ver swagger ser implementado me fez imaginar implementando a API do zero utilizando recursos similares, enfim, uma sugestão.

Obrigado e mais uma vez parabéns a todos envolvidos.

**Comment by gabrielbaltazar** [comment]
@mborgues obrigado pelo Feedback.

O Middleware não é nativo no Horse porque o Horse é fechado para modificação e aberto para extensão.
Ou seja, o mudança do Middleware de documentação swagger não afeta em nada o Core do Horse, e nos dá a possibilidade de termos outros Middlewares de documentação que trabalhem de maneiras diferentes.
O que o time pode avaliar é adotar esse Middleware como oficial, mas colocar ele dentro Core do Horse estaria ferindo o conceito dele.

Grato.

**Reply by mborgues** [reply]
Ah sim, entendido, apenas citei o middleware swagger por conta dos recursos de data annotantions, controllers e a forma que são registrados e utilizados para geração das rotas, validações das models e tal, algo que ficaria bem interessante ver nativo no Horse para construção do projeto da API.
A minha sugestão era para ter algo assim no Horse, onde a controllers e classes pudessem utilizar data annotations nativos, registros das controllers ao invés de declarar os métodos no dpr, algo como já vemos em outras linguagens 
Exemplo: 
![Capturar2](https://user-images.githubusercontent.com/7889957/127902933-1ca82c18-d879-434c-91c3-b513990a3ab5.PNG)

**Reply by viniciussanchez** [reply]
Como o @gabrielbaltazar disse isso não é a ideia do Horse. Com relação a organização e registrar as controllers, nos trabalhamos assim: https://github.com/viniciussanchez/embarcadero-conference/tree/master/2020/backend
E não colocamos todos os endpoints no DPR. Porém isso vai da forma que você queira trabalhar, o Horse é bem versátil pra isso, você trabalha da forma que quer usando o que quer....
